cub/cub/device/dispatch/tuning/tuning_unique_by_key.cuh

```diff
@@ -344,364 +320,273 @@ template <class KeyT,
 };
 
 template <class KeyT, class ValueT>
-struct sm80_tuning<KeyT, ValueT, primitive_key::yes, primitive_val::yes, key_size::_8, val_size::_8>
+struct sm90_tuning<KeyT, ValueT, primitive_key::yes, primitive_val::yes, key_size::_8, val_size::_8>
 {
-  static constexpr int threads = 192;
-
-  static constexpr int items = 7;
-
-  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_DIRECT;
-
-  static constexpr CacheLoadModifier load_modifier = LOAD_DEFAULT;
-
-  using delay_constructor = detail::fixed_delay_constructor_t<764, 1155>;
+  static constexpr int threads                       = 448;
+  static constexpr int items                         = 11;
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_DEFAULT;
+  using delay_constructor                            = detail::no_delay_constructor_t<1190>;
 };
 
 template <class KeyT, class ValueT>
-struct sm80_tuning<KeyT, ValueT, primitive_key::yes, primitive_val::no, key_size::_8, val_size::_16>
+struct sm90_tuning<KeyT, ValueT, primitive_key::yes, primitive_val::no, key_size::_8, val_size::_16>
 {
-  static constexpr int threads = 128;
-
-  static constexpr int items = 7;
-
+  static constexpr int threads                       = 256;
+  static constexpr int items                         = 9;
   static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
-
-  static constexpr CacheLoadModifier load_modifier = LOAD_DEFAULT;
-
-  using delay_constructor = detail::fixed_delay_constructor_t<992, 1135>;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_DEFAULT;
+  using delay_constructor                            = detail::no_delay_constructor_t<1155>;
 };
 
-} // namespace unique_by_key
-} // namespace detail
-
-template <typename KeyInputIteratorT, typename ValueInputIteratorT = unsigned long long int*>
-struct DeviceUniqueByKeyPolicy
+template <class KeyT, class ValueT>
+struct policy_hub
 {
-  using KeyT   = typename std::iterator_traits<KeyInputIteratorT>::value_type;
-  using ValueT = typename std::iterator_traits<ValueInputIteratorT>::value_type;
+  // using KeyT   = typename std::iterator_traits<KeyInputIteratorT>::value_type;
+  // using ValueT = typename std::iterator_traits<ValueInputIteratorT>::value_type;
 
-  // SM350
-  struct Policy350 : ChainedPolicy<350, Policy350, Policy350>
+  template <int NOMINAL_4B_ITEMS_PER_THREAD, int THREADS>
+  struct DefaultPolicy
   {
-    static constexpr int INPUT_SIZE = sizeof(KeyT);
-    enum
-    {
-      NOMINAL_4B_ITEMS_PER_THREAD = 9,
-      ITEMS_PER_THREAD            = Nominal4BItemsToItems<KeyT>(NOMINAL_4B_ITEMS_PER_THREAD),
-    };
-
+    static constexpr int nominal_4b_items_per_thread = NOMINAL_4B_ITEMS_PER_THREAD;
+    static constexpr int ITEMS_PER_THREAD            = Nominal4BItemsToItems<KeyT>(nominal_4b_items_per_thread);
```

Comment:
Please use PascalCase for template parameter names. ALso kill the variable `nominal_4b_items_per_thread` and use the template parameter directly.